src/utils/pageWrappers/withRelay.js

```diff
@@ -12,11 +12,18 @@ const withRelay = (ChildComponent) => {
     const { initialRecords, ...otherProps } = props
     const AuthUser = useAuthUser()
 
+    // Publish initial records to the store only if they've
+    // changed. Otherwise, old initial records might overwrite
+    // new store data.
+    const previousInitialRecords = usePrevious(initialRecords)
+    const publishInitialRecords = previousInitialRecords !== initialRecords
```

Comment:
@kmjennison ok so we're only publishing records if they've changed from what was previously cached.  Just noting this here so we can talk about it, do you know why the cache reloads old data again after we're fetching new data?

Reply:
Yes, let's discuss.